addon/modifiers/scroll-shadow.ts

```diff
@@ -0,0 +1,59 @@
+// @ts-ignore
+import { setModifierManager } from '@ember/modifier';
+
+type ColorAvailable = 'field' | undefined;
+
+type State = {
+  element: Element;
+  infiniteShadowElement: Element | null;
+  resizeObserver: ResizeObserver;
+  hasScrollbar: boolean;
+};
+
+export default setModifierManager(
+  () => ({
+    createModifier() {
+      return {
+        element: null
+      };
+    },
+
+    baseCSSClass() {
+      return 'scroll-shadow';
+    },
+
+    colorCSSClass(args: any) {
+      const color: ColorAvailable = args.named.color;
+      return `scroll-shadow--${color || 'default'}`;
+    },
+
+    installModifier(state: State, element: Element, args: any) {
+      state.element = element;
+
+      state.resizeObserver = new ResizeObserver((entries: ResizeObserverEntry[]) => {
+        window.requestAnimationFrame(() => {
```

Comment:
Without this, tests fail with `ResizeObserver - loop limit exceeded`. I found this fix in https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded